icechunk/src/session.rs

```diff
@@ -484,9 +567,14 @@ impl Session {
         coord: ChunkIndices,
         data: Option<ChunkPayload>,
     ) -> SessionResult<()> {
-        if let NodeData::Array { shape, .. } = node.node_data {
+        if let NodeData::Array { shape, dimension_names, .. } = node.node_data {
             if shape.valid_chunk_coord(&coord) {
-                self.change_set.set_chunk_ref(node.id, coord, data);
+                let splits = self
+                    .get_splits(&node.id, &node.path, &shape, &dimension_names)
+                    // FIXME: this clone is a workaround for two mutable borrows
+                    // on self.change_set
+                    .clone();
```

Comment:
☝🏾 not sure how to avoid this clone

Reply:
I'm not sure that is the problem. I think this has to do with the lifetime of the `splits` argument to `set_chunk_ref`